aemcmc/sample/prior.py

```diff
@@ -0,0 +1,27 @@
+import aesara
+import aesara.tensor as at
+import aesara.tensor.random as ar
+
+
+def sample_prior(
+    srng: ar.RandomStream, num_samples: at.TensorVariable, *rvs: at.TensorVariable
+) -> at.TensorVariable:
+    """Sample from a model's prior distributions.
+
+    Parameters
+    ----------
+    srng:
+        `RandomStream` instance with which the model was defined.
+    num_samples:
+        The number of prior samples to generate.
+    rvs:
+        The random variables whose prior distribution we want to sample.
+
+    """
+
+    def step_fn():
+        return rvs, srng.state_updates
```

Comment:
I think it's possible that `srng.state_updates` could include updates for RNGs that aren't in the `rvs` graphs.  If we can go without returning that, and I think we can, then we probably should.

Reply:
What should we return instead to only get updates for rngs in the `rvs` graph?

Reply:
We probably need an `updates` kwarg that is returned in the body of `step_fn`.  This should cover cases in which a model has a `Scan` with updates.  Speaking of which, we definitely need a test for that.

Otherwise, `scan` should pick up the default updates of properly constructed `RandomVariable`s (as well as any other `SharedVariable`s with default updates).  If it doesn't we need to fix that at the `Scan` level.

Reply:
I've updated the approach so that only the relevant updates in the `RandomStream` object and `SharedVariable.default_update`s are used.  This should prevent the inclusion of unnecessary/disconnected update-induced sampling steps.  (N.B. I also added a test for the aforementioned case.)